### PR TITLE
Add three more options for MFEM_TIMER_TYPE [more-timer-options]

### PR DESCRIPTION
**Pull request by v-dobrev**
The options are:
 * 4 - an improved option for Mac (the new default for Mac),
 * 5 - version based on `gettimeofday`, and
 * 6 - version using `MPI_Wtime`.

 In makefile, add target `print-*` that prints the contents of any makefile variable, `*`.

### CODE REVIEW

File: general/tic_toc.cpp [hunk 1]
```diff
@@ -26,6 +26,13 @@
 #define NOMINMAX
 #include <windows.h>
 #undef NOMINMAX
+#elif (MFEM_TIMER_TYPE == 4)
+#include <ctime>
+#include <mach/mach_time.h>
+#elif (MFEM_TIMER_TYPE == 5)
+#include <sys/time.h> // gettimeofday
+#elif (MFEM_TIMER_TYPE == 6)
```
Comment (tzanio) [inline general/tic_toc.cpp]: should this be inside `#ifdef MFEN-USE_MPI`?

File: makefile [hunk 2]
```diff
@@ -441,3 +441,6 @@ style:
 	@if ! $(ASTYLE) $(FORMAT_FILES) | grep Formatted; then\
 	   echo "No source files were changed.";\
 	fi
+
+print-%: ; @printf "%s:\n" $*
```
Comment (tzanio) [inline makefile]: Delete, or add an example/comment how to use this?

File: general/tic_toc.cpp (hunk 1 above)
Comment (v-dobrev) [inline general/tic_toc.cpp]: I'm fine either way. What do you prefer?

File: makefile (hunk 2 above)
Comment (v-dobrev) [inline makefile]: I'll add an example in a comment.

File: general/tic_toc.cpp (hunk 1 above)
Comment (tzanio) [inline general/tic_toc.cpp]: maybe this makes sense actually even without _hypre_... lets keep it

**Pull request merged**